CONTRIBUTING.md

```diff
@@ -31,7 +31,38 @@ To setup a local development environment we recommend reading [Building Grafana
 
 - Add tests relevant to the fixed bug or new feature.
 
-- Follow [PR and commit messages guidelines](#PR-and-commit-messages-guidelines)
+### High level checks
+
+- [ ] The pull request adds value and the impact of the change is in line with [Backend](https://github.com/grafana/grafana/tree/master/pkg) or [Frontend](https://github.com/grafana/grafana/tree/master/style_guides).
+- [ ] The pull request works the way it says it should do.
+- [ ] The pull request does not increase the Angular code base.
+  > We are in the process of migrating to React so any increment of Angular code is generally discouraged from. (there are a few exceptions)
+- [ ] The pull request closes one issue if possible and does not fix unrelated issues within the same pull request.
+- [ ] The pull request contains necessary tests.
+
+### Low level checks
+
+- [ ] The pull request contains a title that explains the PR. Follow [PR and commit messages guidelines](#Pull-Requests-titles-and-message)
+- [ ] The pull request contains necessary link(s) to issue(s). 
+- [ ] The pull request contains commits with commit messages that are small and understandable. Follow [PR and commit messages guidelines](#Pull-Requests-titles-and-message)
+- [ ] The pull request does not contain magic strings or numbers that could be replaced with an `Enum` or `const` instead.
+
+#### Bug specific checks
+
+- [ ] The pull request contains `Closes: #Issue` or `Fixes: #Issue` in pull request description.
+
+### Frontend specific checks
+
+- [ ] The pull request does not contain uses of `any` or `{}` without comments describing why.
+- [ ] The pull request does not contain large React component that could easily be split into several smaller components.
+- [ ] The pull request does not contain back end calls directly from components, use actions and Redux instead.(there can be exceptions)
+
+#### Redux specific checks (skip if pull request does not contain Redux changes)
```

Comment:
Shouldn't this section be in the frontend style guide?

Reply:
I find the guidelines be great when creating a PR but harder when reviewing it.

Reply:
Not sure I understand. Do you mean these things are written also in style guide but here are written differently which makes it easier to understand when reviewing a PR?

What I mean is that code in a PR should adhere to the style guide so this seems like either a duplication or if it is not duplicated, something that seems appropriate to be near other rules/tips on how to write redux code.

Reply:
Ok I think I understand what your point is, I'll copy this it to Front End Style Guides as well.